### PR TITLE
Add support for Puppet 8, drop support for Puppet 3/4/5

### PR DESCRIPTION
**Pull request by Phil-Friderici**
* update to PDK v3.0.0
* use stdlib v8.0.0
* use rpcbind v3.0.0
* use nfs v2.3.0

### CODE REVIEW

**Comment by Phil-Friderici** [comment]
@anders-larsson from my perspective this PR should be good, please have a look.
I would recommend to also update the dependency modules nfs & rpcbind as well.

**Review by anders-larsson — APPROVED** [review]
Looks good to me. Good job!

**Comment by anders-larsson** [comment]
I'll add these changes to a separate branch to allow testing in a real environment.

File: manifests/init.pp [hunk 1]
```diff
@@ -110,18 +109,18 @@
       require   => $nfs_requires,
     }
 
-    if "${::osfamily}-${::operatingsystemrelease}" =~ /^Suse-11/ {
+    if include_sysconfig {
```
Comment (anders-larsson) [inline manifests/init.pp]: Found a bug!

@Phil-Friderici  We're missing a dollar sign here ($) to indicate it's a variable?

Interesting that the spec tests didn't catch it though. Found it by testing it on a real system and something didn't add up.

**Review by anders-larsson — CHANGES_REQUESTED** [review]
See comment before this one.

**Comment by anders-larsson** [comment]
Works correctly (at least on Ubuntu 22.04) after fixing it locally. Maybe we need to revise the spec tests so this is covered?

File: manifests/init.pp (hunk 1 above)
Comment (Phil-Friderici) [reply manifests/init.pp]: Thanks for finding this ! I've fixed the code and the tests accordingly.
Maybe I should rethink my approach to usually dont check for non existing resources ;)

Have a great weekend!!!

**Comment by Phil-Friderici** [comment]
> Works correctly (at least on Ubuntu 22.04) after fixing it locally. Maybe we need to revise the spec tests so this is covered?

Fixed them by using `is_expected.not_to contain"_resource('name')`

**Comment by anders-larsson** [comment]
Looks good to me.

**Review by anders-larsson — APPROVED** [review]

**Comment by anders-larsson** [comment]
Does the change of type of $gss (not accepting strings) warrant a major release? Strictly speaking I guess it would be the case?

**Comment by Phil-Friderici** [comment]
Thanks for review, released as v1.3.0

**Pull request merged**